### DISCUSSION TITLE
Can my AUX VSP ePump be assigned to a non_Button_xx?

### DISCUSSION THREAD

**Discussion by Bugman1400** [aqualinkd/AqualinkD#344] [General]
For the Web UI to work for my RS-4 Combo, my AUX VSP is currently assigned to the Solar Heater button (Button_08) even though I don't have a Solar Heater. Also, this shows up as an EXTRA AUX device on the OT panel when I use AqualinkD to turn the VSP on. However, when I turn the VSP on via the OT panel the EXTRA AUX device is not indicated. Is it possible to set my AUX VSP to Button_09 in the `/etc/aqualinkd.conf` file so the EXTRA AUX device is not indicated?
![Screenshot from 2024-09-07 09-13-15](https://github.com/user-attachments/assets/decfe268-5602-4c07-bf5d-66ef26552ac2)

**Comment by sfeakes** [comment]
I'm not sure what you mean by `OT panel` , so this may not be 100% what you are looking for.

Solar Heater & Extra Aux are the same thing (ie the same button) Jandy simply renamed it at some point in their history.

If you want it to show as Extra Aux simply use the button label.  ie. `Button_08_label = Extra Aux`, you can call it something more descriptive if you want as well `Button_08_label = Waterfall`.  If you don't want it displayed at all `Button_08_label = NONE`.

If you have your VSP is assigned to that button, then simply add the VSP options to it as well, example below.
```
Button_08_label = AUX Pump
button_08_pumpIndex=1
button_08_pumpID=0x60
button_08_pumpType=Pentair VS
```

AqualinkD works as below with these options.
The on/off state is fixed to a button corresponding to the Jandy control panel. this can not be changed, only the name of the physical button in AqualinkD UI can be changed. This is button_??_label.
All the other pump options are simply the AqualinkD button to assign the VSP pump information to.  You can actually assign them to any button you like, BUT the RPM/GPM/Watts will only be displayed when that button is on.

Button_09 does not exist on a RS4 panel, so no you can't assign that.

**Reply by Bugman1400** [reply]
The OT Panel is the common abbreviation for the One Touch Panel. Also, I think you misinterpreted my post, but perhaps gave me the answer I was looking for.
I believe your main point is that a VSP cannot be assigned a tile in the Web UI without being assigned to one of the Buttons listed above under the RS-4 Combo column. This is unfortunate because I don't think this is actually how the Aqualink RS-4 Combo control panel and OT Panel work. In general, VSPs are powered ON all the time. Their power source is not typically wired to one of the relays in the power panel like a single speed motor, blower, or pool light. Therefore, VSPs are not typically assigned to a Button and the speed cannot be set via the Aqualink RS control panel (as seen below). However, VSP speeds can be set at the OT Panel. I understand that the workaround for AqualinkD is to assign the VSP to a Button until future improvements can be made.

Thank you for the info.

![Screenshot from 2024-09-08 15-00-21](https://github.com/user-attachments/assets/5db6c8a5-c771-478c-9c56-2f712414cebf)

**Reply by sfeakes** [reply]
> In general, VSPs are powered ON all the time.

Now I get your point and why you want it this way.  I'll look at making a change to assign a virtual button to VSP pump.

BTW, VSP not assigned to a circuit that can be turned on/off is not what I have seen from the people using this project, with over 3k people using this it hasn't come up before.  Also seems like a bad idea for many setups. example many VSP are set that when they looses connection to the control panel, they will default to a standard run cycle.  So if you've turned VSP to 0 through the control panel for backwash, maintenance etc, then the pump looses connection, it'll turn on. S

**Reply by Bugman1400** [reply]
> > In general, VSPs are powered ON all the time.
> 
> Now I get your point and why you want it this way. I'll look at making a change to assign a virtual button to VSP pump.
> 
> BTW, VSP not assigned to a circuit that can be turned on/off is not what I have seen from the people using this project, with over 3k people using this it hasn't come up before. Also seems like a bad idea for many setups. example many VSP are set that when they looses connection to the control panel, they will default to a standard run cycle. So if you've turned VSP to 0 through the control panel for backwash, maintenance etc, then the pump looses connection, it'll turn on. S

I think most people have the VSP as their main Filter Pump in which programming to a Button makes sense. I do not. I am also not aware of VSPs that have a default set speed when comm is lost. That seems dangerous. I know the Jandy ePumps do not. They actually go to 0 rpms when comm is lost for around a minute. 

A Virtual Button sounds like a fantastic idea!

**Reply by sfeakes** [reply]
Pentair pumps will default to a predefined state & speed when control is lost, but can’t say I’ve played with Jandy epum much, just enough to decode their protocol. I’ve made the changes to create a virtual button to assign a VSP, all is working in my test setup, after a few days of running I’ll post an update.

**Comment by sfeakes** [comment]
@Bugman1400  I just posted a dev release 2.4.1 that has the virtual button for pumps.  You can use that if you want until an official release is posted.  You'll need to add something similar to below to your `aqualinkd.conf`
```
virtual_button_01_label = Virtual Pump
virtual_button_01_pumpIndex=1
virtual_button_01_pumpType=JANDY ePUMP
virtual_button_01_pumpID=0x78
```
